### DISCUSSION TITLE
Problem with composables?!

### DISCUSSION THREAD

**Discussion by mihail727** [nuxt/nuxt#17914] [Questions] [answered]
I cant understand why is doesn't work.
[Check this](https://stackblitz.com/edit/github-anyypy?file=pages/index.vue)

**Comment by alnaum** [comment, marked as answer]
It seems to work if you change `<button @click="switchVisible">Click</button>` to `<button @click="Menu.switchVisible">Click</button>` in pages/index.vue.

**Reply by mihail727** [reply]
My bad, yes in the example it work..... but that's not the problem
I was too quick to describe the problem...

**Reply by mihail727** [reply]
> It seems to work if you change `<button @click="switchVisible">Click</button>` to `<button @click="Menu.switchVisible">Click</button>` in pages/index.vue.

Check now
[Check this](https://stackblitz.com/edit/github-anyypy?file=pages/index.vue)

when i try to manage menu from different files it stops working

**Reply by alnaum** [reply]
I think it is because each call of useMenu() reinitializes isVisible reference and its value. If you change `const isVisible = ref(false);` to `const isVisible = useState('isVisible', false)` in composables/useMenu.ts it should work. Check [useState](https://v3.nuxtjs.org/docs/usage/state).

**Reply by mihail727** [reply]
> I think it is because each call of useMenu() reinitializes isVisible reference and its value. If you change `const isVisible = ref(false);` to `const isVisible = useState('isVisible', false)` in composables/useMenu.ts it should work. Check [useState](https://v3.nuxtjs.org/docs/usage/state).

YEAP! it works.... thanks
